python/lsst/ts/aos/utils/hexapods/hexapod_lut.py

```diff
@@ -0,0 +1,221 @@
+            fitted_polynomial, [residual, _, _, _] = Polynomial.fit(
+                data_camera[f"target_{lut_type}"],
+                data_camera[field] - np.mean(data_camera[field]),
+                polynomial_degree,
+                full=True,
+            )
+            lut_data[f"{component.lower()}_config"][f"{lut_type}_coeffs"][
+                idx
+            ] = fitted_polynomial.convert().coef.tolist()
+
+    # Save the updated LUT file
+    print("Saving LUT file")
+
+    with open(output_path, "w") as yaml_file:
+        yaml.safe_dump(lut_data, yaml_file, default_flow_style=False, sort_keys=False)
+
+
+def parse_arguments() -> argparse.Namespace:
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(
+        description="Update Hexapod LUT file with Laser Tracker offsets"
+    )
+
+    parser.add_argument(
+        "start_time",
+        help="Start time in a valid format: 'YYYY-MM-DD HH:MM:SSZ'",
+        type=Time,
+    )
+
+    parser.add_argument(
+        "end_time",
+        help="End time in a valid format: \
+            'YYYY-MM-DD HH:MM:SSZ'",
+        type=Time,
+    )
+
+    parser.add_argument(
+        "--lut_path", default=None, help="Path to the original LUT file"
+    )
+
+    parser.add_argument(
+        "--output_path", default="hexapod_lut.yaml", help="Path to the output LUT file"
+    )
+
+    parser.add_argument(
+        "--lut_type",
+        choices=["elevation", "rotation", "azimuth"],
+        default="elevation",
+        help="Type of LUT to be updated",
+    )
+
+    parser.add_argument(
+        "--components",
+        choices=["M2", "Camera", "M2Camera"],
+        default="M2Camera",
+        help="Component to update the LUT for. \
+            Defaults to both M2 and Camera.",
+    )
+
+    parser.add_argument(
+        "--is_compensation_mode_on",
+        type=bool,
+        default=False,
+        help="Was hexapod compensation mode on when taking \
+            the data. If so, LUT update is an increment.",
+    )
+
+    parser.add_argument(
+        "--polynomial-degree",
+        default=5,
+        type=int,
+        help="Degree of the polynomial to be fitted to the data",
+    )
+
+    parser.add_argument(
+        "--efd",
+        default="usdf_efd",
+        help="EFD name. Defaults to usdf_efd",
+    )
+
+    return parser.parse_args()
+
+
+async def process_data(args: argparse.Namespace) -> None:
+    print("Connecting to EFD", args.efd)
+    client = EfdClient(args.efd)
+
+    await update_hexapod_lut(
+        args.start_time,
+        args.end_time,
+        args.lut_type,
+        args.lut_path,
+        args.output_path,
+        args.components,
+        args.is_compensation_mode_on,
+        args.polynomial_degree,
+        client,
+    )
+
+
+def run() -> None:
+    """Main function to run the process_data function."""
+    asyncio.run(process_data(parse_arguments()))
+
+
+if __name__ == "__main__":
```

Comment:
please remove this and make sure you add the script to the [script session](https://github.com/lsst-ts/ts_aos_utils/blob/fe1bcf80fd7994f08aea66cce44586d02265ef73/pyproject.toml#L21) of pyproject.toml to make it available as an executable